text/0038-protocol-api.md

```diff
@@ -0,0 +1,92 @@
+***This RFC is dependent on [RFC 39](0039-service-overhaul.md).***
+
+# Summary
+
+- Add an API to register protocol
+- Protocols are registered with a list of compatible codecs
+- Show only stream encoders compatible with the selected service protocol
+- Add as byproduct the possibility to add plugin with new protocol and own output (useless without RFC 39)
+
+# Motivation
+
+Create a better management of outputs, encoders and services with a protocol centered approach.
+
+Byproduct: Add also the possibility to support third-party plugin which add protocol support with his own output.
+
+*Services addition is a part of [RFC 39](0039-service-overhaul.md).*
+
+# Design
+*This RFC does not consider that because a protocol support a encoder it should be implemented in the main project.*
+
+## Protocols
```

Comment:
Generally making a list of protocols that then have a list of codes seems okay. 

I have some concerns related to how if a protocol is registered service / output plugins configure the protocols. I believe that API layer still needs to be defined.

For example the webrtc protocol will have extra api calls the the service plugin will need to make to drive the process to start the stream.

Reply:
> For example the webrtc protocol will have extra api calls the the service plugin will need to make to drive the process to start the stream.

I think it shall be implemented by the output and service, the protocol is just `"specification" put as code`

Reply:
Many things heave changed since.
- #45